### ISSUE TITLE
`no-thenable` rule with yup conditional validation code

### ISSUE THREAD

**Issue by laozhu** [sindresorhus/eslint-plugin-unicorn#1841]
1️⃣ Explain here what's wrong

When I use [yup](https://github.com/jquense/yup) for conditional validation, there's a keyword `then` keyword in my code, but eslint will throw below warning

```console
Do not add `then` to an object.eslint unicorn/no-thenable
```

 2️⃣ Specify which rule is buggy here and in the title

- [unicorn/no-thenable](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/rules/no-thenable.js)

3️⃣ Add some examples where the issue appears

```ts
import * as yup from 'yup';

const mySchema = yup.object({
  requireUsername: yup.bool(),
  username: yup.string().when('requireUsername', {
    is: true,
    then: yup.string().required('Username is required'),
  }),
})
```

**Comment by fisker** [comment]
Duplicate of #1710, I guess you'll have to use `eslint-disable`
